test/mock/connection.py

```
@@ -0,0 +1,47 @@
+
+    def push(self, message: MessageContract) -> None:
+        if self.broken_push:
+            raise Exception()
+
+
+class BrokenConnectionFactory(ConnectionsFactory):
+
+    def __init__(
+            self,
+            lm: ListenerErrorHandler, qm: QueuingErrorHandler,
+            listener: MessageQueueListener = BrokenListener(), queue: Queue = BrokenQueue(),
```

Ruff (0.12.2)

Do not perform function call `BrokenListener` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)